openedx/core/djangoapps/content_libraries/api/containers.py

```diff
@@ -251,6 +252,25 @@ def delete_container(
     # TODO: trigger a LIBRARY_COLLECTION_UPDATED for each collection the container was in
 
 
+def restore_container(container_key: LibraryContainerLocator) -> None:
+    """
+    Restore the specified library container.
+    """
+    try:
+        container = _get_container(container_key, isDeleted=True)
+    except ContentLibraryContainerNotFound:
+        return
+
```

Comment:
I think we should raise an error if we fail to restore the container.. It's enough to remove this try/catch -- that lets `ContentLibraryContainerNotFound` propagate through to the REST API view, where `convert_exceptions` already handles it as a 404.

```suggestion
    container = _get_container(container_key, isDeleted=True)
```

Reply:
Updated: https://github.com/openedx/edx-platform/pull/36464/commits/06960ff5336d2400762eef76def4e1149c3e8f5a